gcloud-java-dns/src/main/java/com/google/gcloud/dns/spi/DefaultDnsRpc.java

```diff
@@ -193,4 +206,58 @@ public Change getChangeRequest(String zoneName, String changeRequestId, Map<Opti
       throw translate(ex);
     }
   }
+
+  @Override
+  public BatchRequest createBatch() {
```

Comment:
Fine as is for now. However I think we could also have made that Opaque (which would be nicer). Either by using Object (but then rpc impl need to case) or using generics `B`.

If we went that route we could have passed the API level callback to the RPC and let the RPC implementation create its specific callback (in this case `JsonBatchCallback`) which will delegate to the API callback.

I also think that instead of prepareListZones,... returning the Batch we should call it addListZonesToBatch (and return void).